app/src/test/java/org/oppia/android/app/parser/StringToFractionParserTest.kt

```diff
@@ -0,0 +1,296 @@
+      .getErrorMessageFromStringRes(context)
+    assertThat(error)
+      .isEqualTo("None of the numbers in the fraction should have more than 7 digits.")
+  }
+
+  @Test
+  fun testParser_submitTimeError_atLengthLimit_returnsValid() {
+    val error = stringToFractionParser.getSubmitTimeError("1234567/1234567")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo(null)
+  }
+
+  @Test
+  fun testParser_submitTimeError_returnsDivisionByZero() {
+    val error = stringToFractionParser.getSubmitTimeError("123/0")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Please do not put 0 in the denominator")
+  }
+
+  @Test
+  fun testParser_submitTimeError_ambiguousSpacing_returnsInvalidFormat() {
+    val error = stringToFractionParser.getSubmitTimeError("1 2 3/4")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+  }
+
+  @Test
+  fun testParser_submitTimeError_emptyString_returnsInvalidFormat() {
+    val error = stringToFractionParser.getSubmitTimeError("")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+  }
+
+  @Test
+  fun testParser_realTimeError_noNumerator_returnsInvalidFormat() {
+    val error = stringToFractionParser.getRealTimeAnswerError("/3")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+  }
+
+  @Test
+  fun testParser_realTimeError_severalSlashes_returnsInvalidFormat() {
+    val error = stringToFractionParser.getRealTimeAnswerError("1/3/8")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+  }
+
+  @Test
+  fun testParser_realTimeError_severalDashes_returnsInvalidFormat() {
+    val error = stringToFractionParser.getRealTimeAnswerError("-1/-3")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+  }
+
+  @Test
+  fun testParser_parseFractionFromString_incorrectFraction_throwsException() {
+    val exception = assertThrows(IllegalArgumentException::class) {
+      stringToFractionParser.parseFractionFromString("abc")
+    }
+    assertThat(exception)
+      .hasMessageThat()
+      .contains("Incorrectly formatted fraction: abc")
+  }
+
+  @Test
+  fun testParser_parseFraction_incorrectFraction_returnsNull() {
+    val parseFraction = stringToFractionParser.parseFraction("abc")
+    assertThat(parseFraction).isEqualTo(null)
+  }
+
+  @Test
+  fun testParser_parseFraction_parseRegularFraction() {
+    val parseFractionFromString = stringToFractionParser.parseFractionFromString("1/2")
+    val parseFraction = stringToFractionParser.parseFraction("1/2")
+    val constructedFraction = Fraction.newBuilder()
+      .setNumerator(1)
+      .setDenominator(2).build()
```

Comment:
Here & elsewhere: the line-break should happen at the higher syntactical element (e.g. the ``=``) but there's another pattern we're starting to adopt that is a bit cleaner:

```suggestion
    val constructedFraction = Fraction.newBuilder().apply {
      numerator = 1
      denominator  = 2
    }.build()
```

(note that technically we should break at the '=' in this example, too, but I prefer breaking at the '{' which is consistent with our codebase).

I suggest adopting this & using the same syntax spacing. Ditto for other protos constructed in this test file.

Reply:
Updated to use the new syntax.